ddl/table.go

```diff
@@ -152,6 +138,74 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 	}
 }
 
+func getBundleFromDBInfo(t *meta.Meta, job *model.Job, dbInfo *model.DBInfo) (*placement.Bundle, error) {
+	dbBundleWrapper := func(bundle *placement.Bundle) *placement.Bundle {
+		bundle.Index = placement.RuleIndexDatabase
+		bundle.Override = true
+		return bundle
+	}
+	if dbInfo.DirectPlacementOpts != nil {
+		// check the direct placement option compatibility.
+		if err := checkPolicyValidation(dbInfo.DirectPlacementOpts); err != nil {
+			return nil, errors.Trace(err)
+		}
+		// build bundle from direct placement options.
+		bundle, err := placement.NewBundleFromOptions(dbInfo.DirectPlacementOpts)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
```

Comment:
I think if an error happens for when `checkPolicyValidation` or `NewBundleFromOptions` . We should cancel the job, otherwise it will retry infinitely

Reply:
sure